src/main/java/SearchEngine.java

```diff
@@ -0,0 +1,88 @@
+import strategies.impl.AllStrategyImpl;
+import strategies.impl.AnyStrategyImpl;
+import strategies.impl.NoneStrategyImpl;
+import strategies.SearchStrategy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+class SearchEngine {
+
+    private final FileReader fileReader;
+    private final Scanner scanner;
+    private final LineTokenizer lineTokenizer;
+
+
+    public SearchEngine(FileReader fileReader) {
+        this.fileReader = fileReader;
+        scanner = new Scanner(System.in);
+        lineTokenizer = new LineTokenizer();
+    }
+
+    public void run() throws IOException {
+        List<String> peopleList = fileReader.readFile();
+        while (true) {
+            printMenu();
+            String selection = scanner.nextLine();
+            switch (selection) {
+                case "1" -> find(peopleList, lineTokenizer.mapWords(peopleList));
+                case "2" -> printList(peopleList);
+                case "3" -> {
```

Comment:
printMenu() for exit says to use '0', please fix that

Reply:
fixed